tests/test_null_byte_insertion.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 The Matrix.org Foundation C.I.C.
```

Comment:
This file probably wants to be in a more organised directory (and named something a bit more specific to *where* the null byte is being inserted).

There's already `RoomSearchTestCase` in `tests/rest/client/test_rooms.py`, but 1) that file is already big and 2) I'm not sure you're so interested in testing the behaviour from a REST point of view.

My personal feeling would be perhaps `tests/storage/test_room_search.py` (especially if you're going to e.g. call `search_msgs` which is in the `storage` component of Synapse), but I would like to ask @clokep for his opinion.

*Maybe you would* want to test this from the client API point of view and perhaps `tests/rest/client/test_room_search.py` would make sense in that case.

Reply:
> My personal feeling would be perhaps `tests/storage/test_room_search.py` (especially if you're going to e.g. call `search_msgs` which is in the `storage` component of Synapse), but I would like to ask @clokep for his opinion.

This seems reasonable to me. I did ask @H-Shay to look if there are other tests this could be grouped with, but if those are only rest tests then I'm not sure it makes sense to put it with those.